docs/features/tagging/high-level-design.md

```diff
@@ -0,0 +1,147 @@
+
+- **All tag managing operations are done per project**: this limits quite a bit the number of rows that queries must
+  handle, since doing system wide queries are not contemplated.
+- **System wide features are non-taggable**.
+- **Low volume of feature entries**: each project will have features in the hundreds, low thousands at most.
+- **Low number of tag entries**: the initial requirement is to have a max of 1 tag per feature; it is plausible to
+  assume that even in the case that this limitation is expanded upon, the possible number of tags per feature would be
+  in the low tens.
+- **One tag per feature**: queries are further simplified because there won't be querying for tag combinations
+- **The original `features` table remains the same**
+- **Budget**
+
+## Workflow
+
+There are two distinct parts to the tagging system.
+
+### Feature tagging operations
+
+These endpoints deal with handling tags for an individual feature.
+
+- `PATCH /api/v1/geo-features/:featureId/tags`
+
+```typescript
+Payload
+{
+	projectId: string
+	tag: string;
+}
+```
+
+Changes the current tag with the payload's `tag` for the feature with `featureId`.
+`tag` should not be empty and should be validated to not to exceed the configurable maximum number of characters allowed
+for tag names.
+
+In order to maintain capitalization uniqueness, a check is made first to see if there is already a case insensitive
+match with a previously existing tag for the same Project with `projectId`. If so, that tag will be used instead of the
+tag provided in the payload.
+
+- Extend `POST /api/v1/projects/:id/features/shapefile`
+
+The `UploadShapefileDTO` should be expanded to also accept an optional tag field, so that **all Features created from
+the shape file will be tagged with the same tag**.
+
+It should not be empty and should be validated to not to exceed the configurable maximum number of characters allowed
+for tag names.
+
+In order to maintain capitalization uniqueness, a check is made first to see if there is already a case insensitive
+match with a previously existing tag for the same Project with `projectId`. If so, that tag will be used instead of the
+tag provided in the payload.
+
+- `DELETE /api/v1/geo-features/:featureId/tags/`
+
+Removes the tag for the given `featureId`. This means removing the corresponding row on the `feature_tags` table.
+
+## Tag Managing
+
+These endpoints deal with tag managing operations on a per Project level.
+
+- `GET /api/v1/projects/:projectId/tags?tag=someName`
+
+For the given `projectId`, returns a list of all the DISTINCT tags from the Project's Features that partially matches
+the `tag` query parameter. By default, the matching is `containing`, but other options could be considered.
+
+The `tag` query parameter may be optional, so that if it is empty, all the DISTINCT tags for the given `projectId` are
+returned.
+
+Ordered alphabetically by default.
```

Comment:
Will be possible other sorting values? Like `most-recent` or similar. Per design, the default view when the user is about to type a tag is displaying the most recent tags introduced, so it would be ideal having this use case as well.

<img width="549" alt="image" src="https://github.com/Vizzuality/marxan-cloud/assets/999124/9e18cfa5-84a3-4a98-ae25-cda8672ffb95">